test/integration/kubernetes_deploy_test.rb

```diff
@@ -635,6 +635,29 @@ def test_bad_container_on_daemon_sets_fails
     ], in_order: true)
   end
 
+  def test_resource_quotas_are_deployed_first
+    forced_timeout = 10
+    KubernetesDeploy::Deployment.any_instance.stubs(:timeout).returns(forced_timeout)
+    result = deploy_fixtures("resource-quota")
+    assert_deploy_failure(result)
+    assert_logs_match_all([
+      "Predeploying priority resources",
+      "Deploying ResourceQuota/resource-quotas (timeout: 30s)",
+      "Deployment/web deployment timed out",
+      "Successful resources",
+      "ResourceQuota/resource-quotas",
+      "Deployment/web: TIMED OUT (limit: 10s)",
+      "failed quota: resource-quotas"
+    ], in_order: true)
+
+    rqs = kubeclient.get_resource_quotas(namespace: @namespace)
+    assert_equal 1, rqs.length
+
+    rq = rqs[0]
+    assert_equal "resource-quotas", rq["metadata"]["name"]
+    assert rq["spec"].present?
```

Comment:
nbd but it'd be nice to verify the success condition a little more specifically here, e.g. compare spec.hard.pods to status.hard.pods or something.

Reply:
That gets checked during the success deploy though. I can do that here too but I thought just making sure the spec was present would be useful enough to check for existence